model/test_results.go

```diff
@@ -348,16 +392,35 @@ func (t *TestResults) downloadParquet(ctx context.Context) ([]TestResult, error)
 		return nil, catcher.Resolve()
 	}
 
-	pr, err := reader.NewParquetReader(buffer.NewBufferFileFromBytes(data), new(ParquetTestResults), 1)
+	fr, err := goparquet.NewFileReader(bytes.NewReader(data))
 	if err != nil {
 		return nil, errors.Wrap(err, "creating Parquet reader")
 	}
+	pr := floor.NewReader(fr)
+	var readerClosed bool
+	defer func() {
+		if !readerClosed {
+			_ = pr.Close()
+		}
+	}()
+
+	var parquetResults []ParquetTestResults
+	for pr.Next() {
+		row := ParquetTestResults{}
+		if err := pr.Scan(&row); err != nil {
+			return nil, errors.Wrap(err, "reading Parquet test results row")
+		}
 
-	parquetResults := make([]ParquetTestResults, pr.GetNumRows())
-	if err := pr.Read(&parquetResults); err != nil {
-		return nil, errors.Wrap(err, "unmarshalling Parquet test results")
+		parquetResults = append(parquetResults, row)
+	}
+
+	if err := pr.Err(); err != nil {
+		return nil, errors.Wrap(err, "reading Parquet test results rows")
+	}
+	readerClosed = true
+	if err := pr.Close(); err != nil {
+		return nil, errors.Wrap(err, "closing Parquet reader")
```

Comment:
could it make sense to just log the error in this case? We already have the results

Reply:
Good idea! I went overboard here, fixed.